tests/snmp/test_snmp_phy_entity.py

```diff
@@ -0,0 +1,583 @@
+            else:
+                parent_oid = MODULE_TYPE_FAN_DRAWER + parent_position * MODULE_INDEX_MULTIPLE
+        expect_oid = parent_oid + DEVICE_TYPE_FAN + position * DEVICE_INDEX_MULTIPLE
+        assert expect_oid in snmp_physical_entity_info, 'Cannot find fan {} in physical entity mib'.format(name)
+        fan_snmp_fact = snmp_physical_entity_info[expect_oid]
+        assert fan_snmp_fact['entPhysDescr'] == name
+        assert fan_snmp_fact['entPhysContainedIn'] == CHASSIS_SUB_ID if parent_name == CHASSIS_KEY else parent_oid
+        assert fan_snmp_fact['entPhysClass'] == PHYSICAL_CLASS_FAN
+        assert fan_snmp_fact['entPhyParentRelPos'] == position
+        assert fan_snmp_fact['entPhysName'] == name
+        assert fan_snmp_fact['entPhysHwVer'] == ''
+        assert fan_snmp_fact['entPhysFwVer'] == ''
+        assert fan_snmp_fact['entPhysSwVer'] == ''
+        assert fan_snmp_fact['entPhysSerialNum'] == '' if is_null_str(fan_info['serial']) else fan_info[
+            'serial']
+        assert fan_snmp_fact['entPhysMfgName'] == ''
+        assert fan_snmp_fact['entPhysModelName'] == '' if is_null_str(fan_info['model']) else fan_info['model']
+        assert fan_snmp_fact['entPhysIsFRU'] == REPLACEABLE if fan_info[
+                                                                   'is_replaceable'] == 'True' else NOT_REPLACEABLE
+
+        if not is_null_str(fan_info['speed']):
+            tachometers_oid = expect_oid + SENSOR_TYPE_FAN
+            assert tachometers_oid in snmp_physical_entity_info, 'Cannot find fan tachometers info in physical entity mib'
+            tachometers_fact = snmp_physical_entity_info[tachometers_oid]
+            assert tachometers_fact['entPhysDescr'] == 'Tachometers for {}'.format(name)
+            assert tachometers_fact['entPhysContainedIn'] == expect_oid
+            assert tachometers_fact['entPhysClass'] == PHYSICAL_CLASS_SENSOR
+            assert tachometers_fact['entPhyParentRelPos'] == 1
+            assert tachometers_fact['entPhysName'] == 'Tachometers for {}'.format(name)
+            assert tachometers_fact['entPhysHwVer'] == ''
+            assert tachometers_fact['entPhysFwVer'] == ''
+            assert tachometers_fact['entPhysSwVer'] == ''
+            assert tachometers_fact['entPhysSerialNum'] == ''
+            assert tachometers_fact['entPhysMfgName'] == ''
+            assert tachometers_fact['entPhysModelName'] == ''
+            assert tachometers_fact['entPhysIsFRU'] == NOT_REPLACEABLE
+
+
+def test_psu_info(duthost, snmp_physical_entity_info):
+    """
+    Verify PSU information in physical entity mib with redis database
+    :param duthost: DUT host object
+    :param snmp_physical_entity_info: Physical entity information from snmp fact
+    :return:
+    """
+    keys = redis_get_keys(duthost, STATE_DB, PSU_KEY_TEMPLATE.format('*'))
+    if not keys:
+        pytest.skip('PSU information not exists in DB, skipping this test')
```

Comment:
>pytest.skip [](start = 8, length = 11)

There are many `pytest.skip()`. Seems are a valid failure case.
Do you skip just for backward-compatible? If yes, I think this is not a good method.

Reply:
I skip this for vsimage test. I found this testcase will be ran in vsimage, and there is no PSU information in DB.

Reply:
@wangxin What is best practice to skip test case in vsimage test?

---
In reply to: [532159260](https://github.com/Azure/sonic-mgmt/pull/2379#discussion_r532159260) [](ancestors = 532159260)

Reply:
You can mark the test case like below to specify that this test case is only applicable for physical testbed, not for vsimage.
```
@pytest.mark.device_type('physical')
def test_xxx
```